src/pages/MapViewer/components/Toolbox/components/ToolboxItems/index.tsx

```diff
@@ -37,28 +45,49 @@ const ToolboxItems = () => {
       </div>
 
       <div className="toolbox__items">
-        {selectedCollectionItems?.features?.map((item: Feature) => {
+        {selectedCollectionItems?.features?.map((item: StacFeature) => {
           return (
             <ToolboxRow
               key={item.id}
               dataPoints={parseFeatureDataPoints(item)}
               thumbnail={returnFeatureThumbnail(item)}
               title={item.id.toString()}
               onClick={(e) => {
-                // Temporary function to add layer to the map
-                map.eachLayer((layer) => {
-                  if (layer instanceof L.GeoJSON) {
-                    map.removeLayer(layer);
-                  }
+                const polygon = new Feature({
+                  geometry: new Polygon(item.geometry.coordinates),
+                  name: item.id,
+                });
+
+                // Reproject the geometry from EPSG:4326 to what is used by the map i.e. EPSG:3857
+                polygon.getGeometry().transform(DATA_PROJECTION, MAP_PROJECTION);
+                const bufferedExtent = buffer(polygon.getGeometry().getExtent(), 0.1);
+
+                const vectorSource1 = new VectorSource({
+                  features: [polygon],
                 });
-                const boundsLayer = L.geoJSON(item.geometry);
-                boundsLayer.setStyle({
-                  color: 'red',
-                  weight: 2,
-                  fillOpacity: 0,
+
+                const vectorLayer = new VectorLayer({
+                  source: vectorSource1,
+                  style: new Style({
+                    stroke: new Stroke({
+                      color: 'blue',
```

Comment:
In the original design, the stroke was red and had no fill colour.

Reply:
Okay, I personally like a fill and red is generally reserved as an error color, but I'm happy to change that.

Reply:
I've updated the styling now to match what it was previously, don't really remember why I changed it, likely copy/paste from something similar I had from another project.